pkgs/development/python-modules/cyclonedds-python/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+, cyclonedds
+, setuptools
+, rich-click
+
+, pytest
+, pytest-cov
+, pytest-mock
+, pytest-asyncio
+}:
+
+buildPythonPackage rec {
+  pname = "cyclonedds-python";
+  version = "0.10.5";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "eclipse-cyclonedds";
+    repo = pname;
+    rev = version;
+    hash = "sha256-MN3Z5gqsD+cr5Awmsia9+uCHL/a2KQP2uMS13rVc1Hw=";
+  };
+
+  nativeBuildInputs = [ setuptools ];
+
+  buildInputs = [ cyclonedds ];
+
+  pythonPath = [ rich-click ];
+
+  checkInputs = [
+    pytest
+    pytest-cov
+    pytest-mock
+    pytest-asyncio
+  ];
+
+  env.CYCLONEDDS_HOME = "${cyclonedds.out}";
+
+  pipInstallFlags = [ "--no-binary" ];
```

Comment:
```suggestion
```

The build process is not using pip, since you enabled the PEP517 based build process through `pyproject = true`. It uses `pyproject-hooks`, `build` and `installer`.